core-bundle/src/EventListener/DbafsMetadataSubscriber.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Event\AbstractDbafsMetadataEvent;
+use Contao\CoreBundle\Event\ContaoCoreEvents;
+use Contao\CoreBundle\Event\RetrieveDbafsMetadataEvent;
+use Contao\CoreBundle\Event\StoreDbafsMetadataEvent;
+use Contao\CoreBundle\File\Metadata;
+use Contao\Image\ImportantPart;
+use Contao\StringUtil;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+class DbafsMetadataSubscriber implements EventSubscriberInterface
```

Comment:
I thought about the serializer as well. But IMHO this is not about serialization but "only" the normalization/denormalization part because it's always tied to a database. The serializer is probably overkill for this.

> I would consider making this one event because imho the 99% case is you need both, the serialization and unserialization part.

Yes, you would probably need both. But how would you differentiate? One time you should be able to set the metadata, one time the row data.

Reply:
Hmm. True^^ Didn't think about it properly, sorry.